### ISSUE TITLE
[Bug]: Make session persist by default (docker run)

### ISSUE THREAD

**Issue by enyst** [All-Hands-AI/OpenHands#5515]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

The behavior we currently have, when you use docker and restart, is that it loses the session, it starts a new session. This is good enough for toy projects, not a lot more. The option is to use the dev setup, but the docker run exists precisely to make it easier to use.

Downloading the zip, mounting a local folder, and GitHub tight integration can help mitigate, because hopefully the user will use any of them to save their work from time to time. It still loses the agent's session, thus whatever explanations/context you had the agent. I do think we should fix it. As @rbren pointed out in the issue below, with multiple sessions, we need to anyway.

Refs:
- https://github.com/All-Hands-AI/OpenHands/pull/5317
- https://github.com/All-Hands-AI/OpenHands/pull/5353

### OpenHands Installation

Docker command in README

### Operating System

None

**Comment by rbren** [comment]
Yeah we will definitely handle this in the multi-session world. I will try and think if there's a good way to do this ahead of time, but we're not far out from multi-session

**Comment by mamoodi** [comment]
Did any of this change with the PR merge a few days ago?

**Comment by clmarshall** [comment]
Until then can you tell us how to do it manually?  
The docs are very vague and don't point us to the location of the file to change.

**Comment by rbren** [comment]
I believe this is the new default behavior

**Comment by rbren** [comment]
Actually--if you restart the whole app, maybe not :/

This will require some more thought--we'll need some persistent data to keep a mapping from convo ID <-> runtime ID

**Comment by enyst** [comment]
Why don't we map a volume for the local file store, are there downsides to that?

**Comment by enyst** [comment]
Ah, I see. It was, but it was bugged.

**Comment by mamoodi** [comment]
I'd like to understand what part we are missing right now. Each conversation has its own url now right? So as long as you use the same url, it should retain the session data.

However, even with the current changes if you restart the app, then go to the same URL, the data is not preserved. Is that the issue?

**Comment by mamoodi** [comment]
@enyst we should be able to close this right? Development and Docker both now persist the session by default.

**Comment by enyst** [comment]
I think so! Any further problems should be their own issue.
